conan_provider.cmake

```diff
@@ -14,18 +14,51 @@ function(detect_os OS OS_VERSION)
         else()
             set(${OS} ${CMAKE_SYSTEM_NAME} PARENT_SCOPE)
         endif()
+        if(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
+            string(REPLACE "-" ";" PLATFORM_LIST ${ANDROID_PLATFORM})
+            list(GET PLATFORM_LIST 1 _OS_API_LEVEL)
+            message(STATUS "CMake-Conan: android_platform=${ANDROID_PLATFORM}")
+            set(${OS_API_LEVEL} ${_OS_API_LEVEL} PARENT_SCOPE)
+        endif()
+    endif()
+endfunction()
+
+
+function(detect_arch OS ARCH)
+    if(${OS} STREQUAL "Android")
+        if(ANDROID_ABI STREQUAL arm64-v8a)
```

Comment:
Shouldn't you use `CMAKE_ANDROID_ARCH_ABI`? As this is in CMake as of version 3.7.

Reply:
From what I've seen, both work to generate the proper build, both check for valid values. It seems slightly more intuitive to use to use `ANDROID_ABI` since it's the [documented input parameter](https://developer.android.com/ndk/guides/cmake). What do you think the advantage would be in using `CMAKE_ANDROID_ARCH_ABI` over `ANDROID_ABI`?

Reply:
As at that location and at https://cmake.org/cmake/help/latest/variable/CMAKE_ANDROID_ARCH_ABI.html
`CMAKE_ANDROID_ARCH_ABI` is described as the actual CMake variable that is used. But I don't know much of Android builds.

Following when I look at the toolchain for Android:
https://android.googlesource.com/platform/ndk/+/master/build/cmake/android.toolchain.cmake
there are the following lines:
```
# Determine the ABI.
if(NOT CMAKE_ANDROID_ARCH_ABI)
  if(ANDROID_ABI STREQUAL "armeabi-v7a with NEON")
    set(CMAKE_ANDROID_ARCH_ABI armeabi-v7a)
  elseif(ANDROID_ABI)
    set(CMAKE_ANDROID_ARCH_ABI ${ANDROID_ABI})
  elseif(ANDROID_TOOLCHAIN_NAME MATCHES "^arm-linux-androideabi-")
    set(CMAKE_ANDROID_ARCH_ABI armeabi-v7a)
  elseif(ANDROID_TOOLCHAIN_NAME MATCHES "^aarch64-linux-android-")
    set(CMAKE_ANDROID_ARCH_ABI arm64-v8a)
  elseif(ANDROID_TOOLCHAIN_NAME MATCHES "^x86-")
    set(CMAKE_ANDROID_ARCH_ABI x86)
  elseif(ANDROID_TOOLCHAIN_NAME MATCHES "^x86_64-")
    set(CMAKE_ANDROID_ARCH_ABI x86_64)
  else()
    set(CMAKE_ANDROID_ARCH_ABI armeabi-v7a)
  endif()
endif()
```
So also in the toolchain the `CMAKE_ANDROID_ARCH_ABI` is set...

Reply:
Then it would seem that ``CMAKE_ANDROID_ARCH_ABI`` is the more general one, to use both with and without the AndroidNDK toolchain.cmake?

Reply:
Yeah, that's a compelling reason, I updated it to use `CMAKE_ANDROID_ARCH_ABI`, thanks for the feedback, @CedricSchmeits and @memsharded !